core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -188,13 +209,16 @@ private[spark] class SparkUnsupportedOperationException(
 private[spark] class SparkClassNotFoundException(
     errorClass: String,
     errorSubClass: Option[String] = None,
-    messageParameters: Array[String],
+    messageParameters: Map[String, String],
```

Comment:
This is tested by QueryExecutionErrorsSuite "INCOMPATIBLE_DATASOURCE_REGISTER: create table using an incompatible data source"